src/main/java/teammates/storage/api/InstructorsDb.java

```diff
@@ -393,8 +393,19 @@ public boolean hasInstructorsForGoogleId(String googleId) {
         return !getInstructorsForGoogleIdQuery(googleId).keys().list().isEmpty();
     }
 
-    private Query<Instructor> getInstructorsForGoogleIdQuery(String googleId) {
-        return load().filter("googleId =", googleId);
+    private Query<Instructor> getInstructorsForGoogleIdQuery(String googleIdOrEmail) {
+        String email;
+        String googleId;
+        if (googleIdOrEmail.toLowerCase().contains("@")) {
```

Comment:
The split should be on `@gmail.com`, not just `@`

Reply:
To confirm, say if the `googleIdOrEmail` is a gmail, e.g. `abc@gmail.com`, both `abc` and `abc@gmail.com` should be queried; if the `googleIdOrEmail` is a non-gmail, e.g. `test@example.com`, we can just query for `test@example.com`, right?

Reply:
Yes. The latter has no ambiguity, so to speak.